### ISSUE TITLE
Shared Library build

### ISSUE THREAD

**Issue by vlazzarini** [libsndfile/libsndfile#987]
libsndfile-1.2.2 sources from release page, built on MacOS
-------------

It appears the CMake configuration build of libsndfile has changed recently and shared libraries are not built by default.

I was wondering why this change was introduced. It seems to be confusing for us who were used to updating our libsndfile from sources and getting a dynamic library. This causes the problem that programs
depending on libsndfile stop building because of link errors to libsndfile dependencies.  

In addition to this, libsndfile.dylib seems to be installed as a link to the static library, which is (AFAIK) non-standard on MacOS (.dylib are dynamic libraries). I think this is definitely a bug in the build system and it needs to be fixed.

But really, it would be good if BUILD_SHARED_LIBS was set to default to true so we can forget the whole issue.

**Comment by evpobr** [comment]
Hi @vlazzarini. I don't recall any such changes.

We're just using the standard CMake behavior. And I'm not going to change it, because not all platforms support shared libraries. It makes sense that CMake chooses the safe `BUILD_SHARED_LIBS` value itself, but you can always change it if you wish.

**Comment by vlazzarini** [comment]
I went to check and I realise what happened. The last build I have here was with 1.0.34 and this was done with `./configure` , which builds a dynamic library. I see that the autoconf build has now been dropped and this is probably the change I am experiencing. 

It is of course your call to keep to the defaults, however I think that in systems that support dynamic libs, developers tend to expect that these would build by default, esp. since that has been the behaviour for upstream sources for a very long time. Checking for this would be an improvement. 

In any case, I was also checking whether those links from libsndfile.dylib to libsndfile.a were being made by `make install` and I could not see any evidence of it. I had seen that in someone else's computer but I can't reproduce it now.

Closing this now, but keeping the request as a suggestion.
